File: python/src/wdl_model/model/statements/wdl_statement.py
"""Base statement nodes for the Python WDL object model."""

from __future__ import annotations

from enum import Enum

from wdl_model.model.base import WdlNode


class ComponentType(Enum):
    """High-level statement families used for traversal and validation dispatch."""

    DECLARATION = "DECLARATION"
    CALL = "CALL"
    CONDITIONAL = "CONDITIONAL"
    SCATTER = "SCATTER"


class WdlStatement(WdlNode):
    """Base interface for workflow and task statements."""

    def componentType(self) -> ComponentType:
        """Return the broad statement family for the current node."""
        raise NotImplementedError

File: typescript/src/model/statements/wdl-statement.ts
/** Base workflow and task statement abstractions. */
import type { WdlNode } from '../base/wdl-node.js';

/** High-level statement families used by traversal and validation code. */
export enum WdlStatementComponentType {
  CALL = 'CALL',
  CONDITIONAL = 'CONDITIONAL',
  DECLARATION = 'DECLARATION',
  SCATTER = 'SCATTER',
}

/** Base interface implemented by WDL statement nodes. */
export interface WdlStatement extends WdlNode {
  /** Returns the broad statement family for this node. */
  componentType(): WdlStatementComponentType;
}
